src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -71,105 +78,135 @@ const NavHeader = (props) => {
+                paddingTop: insets.top,
+                paddingRight: dimen.normalizeDimen(20),
+                zIndex: 2,
+                position: 'absolute',
+                right: 0
+              })
+            }>
+            {!isFollow && isHeaderHide ? (
+              <ButtonFollow handleSetFollow={handleFollowTopic} />
+            ) : (
+              <TouchableOpacity onPress={onShareCommunity} style={styles.shareIconStyle}>
+                <ShareIconCircle color="black" width={32} height={32} />
+              </TouchableOpacity>
+            )}
+          </View>
         </View>
-      </View>
-      <Animated.View style={[styles.headerContainer]}>
-        {!isHeaderHide ? (
+        {!hasSearch && (
           <>
-            <Animated.View style={{opacity: opacityHeaderAnimation}}>
-              <FastImage
-                source={topicDetail?.icon_path ? {uri: topicDetail?.icon_path} : TopicDefaultIcon}
-                style={styles.image}
-              />
-            </Animated.View>
-            <View style={styles.containerAction}>
+            <View style={[styles.headerContainer]}>
               <Animated.View style={{opacity: opacityHeaderAnimation}}>
-                {isFollow ? (
-                  <ButtonFollowing handleSetUnFollow={handleFollowTopic} />
-                ) : (
-                  <ButtonFollow handleSetFollow={handleFollowTopic} />
-                )}
+                <FastImage
+                  source={
+                    initialData?.channelPicutre
+                      ? {uri: initialData?.channelPicutre}
+                      : topicDetail?.icon_path
+                      ? {uri: topicDetail?.icon_path}
+                      : TopicDefaultIcon
+                  }
+                  style={styles.image}
+                />
               </Animated.View>
+              <View style={styles.containerAction}>
+                <Animated.View style={{opacity: opacityHeaderAnimation}}>
+                  {isFollow === undefined && isLoading ? (
+                    <Shimmer width={normalize(100)} height={normalize(36)} />
+                  ) : isFollow ? (
+                    <ButtonFollowing handleSetUnFollow={handleFollowTopic} />
+                  ) : (
+                    <ButtonFollow handleSetFollow={handleFollowTopic} />
+                  )}
+                </Animated.View>
+              </View>
             </View>
+            <Animated.View
+              style={[
+                styles.domain(isHeaderHide),
+                {bottom: normalize(20), opacity: opacityHeaderAnimation}
+              ]}>
+              <TopicDomainHeader {...props} />
+            </Animated.View>
           </>
-        ) : null}
-      </Animated.View>
-      <View
-        onLayout={onDomainLayout}
-        style={[styles.domain(isHeaderHide), {bottom: getBottomPostition()}]}>
-        <TopicDomainHeader {...props} />
+        )}
       </View>
 
       {hasSearch && (
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are several changes in this hunk, including the introduction of a new `Shimmer` component for loading states, the use of `FastImage` for image rendering, and various changes to styles and structure. 

One potential issue is the lack of error handling for the `FastImage` component at lines 108-115. If the image source fails to load (for example, if `initialData?.coverImage` or `coverPath` is null or an invalid URL), there may be no fallback or error state. Consider adding error handling or a default image in case the specified source cannot be loaded.

Another point to note is the zIndex usage at lines 134 and 155. While zIndex works on iOS, it can have inconsistent behavior on Android depending on the version. If this app is also intended for Android, please test thoroughly to ensure the zIndex is working as expected.



<!-- This is an auto-generated comment by OSS CodeRabbit -->